threescale_api/resources.py

```diff
@@ -745,7 +745,7 @@ def proxy(self) -> 'Proxies':
 
     @property
     def mapping_rules(self) -> 'MappingRules':
-        return MappingRules(parent=self, instance_klass=MappingRule)
+        return self.proxy.list().mapping_rules
```

Comment:
@mkudlej Why did you introduce this change?

Reply:
@jsmadis Because of this:
```
[4] ser.mapping_rules.list()
[RES] Response(404): b'<!DOCTYPE html>\n<html lang="en">\n  <head>\n    <meta charset="utf-8" />\n    <title>Not Found</title>\n    <link href="/assets/error.css" media="screen" rel="stylesheet" type="text/css" />\n  </head>\n\n  <body>\n    <div id="wrapper">\n      <div id="content">\n        <h1 class=\'logo-3scale logo-3scale--svg logo-3scale--boxed\' id="logo">\n          3scale\n        </h1>\n\n        <h1>Not Found</h1>\n        <p>Sorry. We can\'t find what you\'re looking for.</p>\n      </div>\n\n      <div id="footer">\n        <ul>\n          <li class="powered-by-3scale last">\n            <a href="http://www.3scale.net" target="_blank">\n              Powered by <span class="logo-3scale logo-3scale--svg logo-3scale--powered">3scale</span>\n            </a>\n          </li>\n        </ul>\n      </div>\n    </div>\n  </body>\n</html>\n'
---------------------------------------------------------------------------
ApiClientError                            Traceback (most recent call last)
<ipython-input-4-e5f5d5a9804f> in <module>()
----> 1 ser.mapping_rules.list()

~/work/3scale/mkudlej-3scale-api-python/threescale_api/defaults.py in list(self, **kwargs)
     69         """
     70         log.info(self._log_message("[LIST] List", args=kwargs))
---> 71         instance = self._list(**kwargs)
     72         return instance
     73 

~/work/3scale/mkudlej-3scale-api-python/threescale_api/defaults.py in _list(self, **kwargs)
    233         """
    234         url = self._entity_url()
--> 235         response = self.rest.get(url=url, **kwargs)
    236         instance = self._create_instance(response=response, collection=True)
    237         return instance

~/work/3scale/mkudlej-3scale-api-python/threescale_api/client.py in get(self, *args, **kwargs)
    211 
    212     def get(self, *args, **kwargs):
--> 213         return self.request('GET', *args, **kwargs)
    214 
    215     def post(self, *args, **kwargs):

~/work/3scale/mkudlej-3scale-api-python/threescale_api/client.py in request(self, method, url, path, params, headers, throws, **kwargs)
    207         response = requests.request(method=method, url=full_url, headers=headers,
    208                                     params=params, verify=self._ssl_verify, **kwargs)
--> 209         process_response = self._process_response(response, throws=throws)
    210         return process_response
    211 

~/work/3scale/mkudlej-3scale-api-python/threescale_api/client.py in _process_response(cls, response, throws)
    234             log.error(message)
    235             if throws:
--> 236                 raise errors.ApiClientError(response.status_code, response.content)
    237 
    238         return response

ApiClientError: Response(404): b'<!DOCTYPE html>\n<html lang="en">\n  <head>\n    <meta charset="utf-8" />\n    <title>Not Found</title>\n    <link href="/assets/error.css" media="screen" rel="stylesheet" type="text/css" />\n  </head>\n\n  <body>\n    <div id="wrapper">\n      <div id="content">\n        <h1 class=\'logo-3scale logo-3scale--svg logo-3scale--boxed\' id="logo">\n          3scale\n        </h1>\n\n        <h1>Not Found</h1>\n        <p>Sorry. We can\'t find what you\'re looking for.</p>\n      </div>\n\n      <div id="footer">\n        <ul>\n          <li class="powered-by-3scale last">\n            <a href="http://www.3scale.net" target="_blank">\n              Powered by <span class="logo-3scale logo-3scale--svg logo-3scale--powered">3scale</span>\n            </a>\n          </li>\n        </ul>\n      </div>\n    </div>\n  </body>\n</html>\n'

```